src/data_hub/dataframe.py

```diff
@@ -31,6 +34,46 @@ class DataFrame(pd.DataFrame):
     _modified_columns: dict = dict()
     """if data is modified in a dataframe, and auto_sync is False, this list will contain the columns that have been modified so that the Deep Origin database can be updated. If an empty list, the Deep Origin database will not be updated, and the dataframe matches the Deep Origin database at the time of creation."""
 
+    _allow_adding_rows: bool = True
```

Comment:
the reason for disallowing adding new rows is as follows:

let's say you have a DB with 10 rows. you then slice into the first 2. at that point, those other 8 rows are still there in the DB -- so you should not be allowed to add a row that interferes with those 8 rows

Reply:
I don't follow -- how would adding a new row interfere with the other 8?